### DISCUSSION TITLE
Yi Outdoor 1080p IFUS SD Free Space option :  Auto video cleaning crash ?

### DISCUSSION THREAD

**Discussion by cbroissa-github** [roleoroleo/yi-hack-Allwinner-v2#299] [Q&A]
Hello,

I use the SD Free Space option in Auto video cleaning, I put 10%.
Now my Free SD Space is 90% and I can't change any option, I have the impression that all is blocked.
No more video are recorded.

I think the option is inverted no ?

Thank you.

**Comment by cbroissa-github** [comment]
I put 0 from the ssh access and reboot, and I can change some options but after a few minutes, I can't change anything et can't delete video recorded.

**Reply by roleoroleo** [reply]
> I put 0 from the ssh access and reboot, and I can change some options but after a few minutes, I can't change anything et can't delete video recorded.

If you put 0 the script shouldn't start.
Probably there is another problem.

**Reply by cbroissa-github** [reply]
You think I can log something when I delete a folder and nothing is deleted ?

**Reply by cbroissa-github** [reply]
I think there is a right problem with a record directory

![2021-11-13_093810](https://user-images.githubusercontent.com/83312047/141612109-85656046-c591-44c2-b78e-65bcb7cd0e06.jpg)
![2021-11-13_093848](https://user-images.githubusercontent.com/83312047/141612111-d53c0548-156c-434d-b043-4973ba68e9b2.jpg)

**Reply by roleoroleo** [reply]
Try to reformat the sd card.

**Reply by cbroissa-github** [reply]
indeed there is a small problem :)
![2021-11-13_143738](https://user-images.githubusercontent.com/83312047/141645924-01c1bf7f-2e02-402b-9fe7-0ba0f969f990.jpg)

**Reply by cbroissa-github** [reply]
I reformat the card and it's OK now.
I edit the discussion title.
Maybe when I was at 10%, the auto video cleaning script made the I/O problem ?

**Reply by roleoroleo** [reply]
Probably some sudden reboot caused the file system to break.
At this point, the behavior of the script is unpredictable.

**Comment by roleoroleo** [comment]
I checked the script:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/blob/master/src/static/static/yi-hack/script/clean_records.sh
I think it's ok.
- The argument passed is the free space.
- The script calculates the space used: `USED_SPACE_LIMIT=$((100-$1))`
- And it deletes files until this condition is true: `while [ "$USED_SPACE" -gt "$USED_SPACE_LIMIT" ]`
